src/simplewallet/simplewallet.cpp

```diff
@@ -2386,6 +2388,43 @@ bool simple_wallet::stop_mining_for_rpc(const std::vector<std::string> &args)
   return true;
 }
 
+bool simple_wallet::show_qr_code(const std::vector<std::string> &args)
+{
+  uint32_t subaddress_index = 0;
+  if (args.size() >= 1)
+  {
+    if (!string_tools::get_xtype_from_string(subaddress_index, args[0]))
+    {
+      fail_msg_writer() << tr("invalid count: must be an unsigned integer");
+      return true;
+    }
+    if (subaddress_index >= m_wallet->get_num_subaddresses(m_current_subaddress_account))
+    {
+      fail_msg_writer() << tr("<subaddress_index> is already out of bound");
+      return true;
+    }
+  }
+
+  const std::string address = "monero:" + m_wallet->get_subaddress_as_str({m_current_subaddress_account, subaddress_index});
+  const qrcodegen::QrCode qr = qrcodegen::QrCode::encodeText(address.c_str(), qrcodegen::QrCode::Ecc::LOW);
+  for (int y = -2; y < qr.getSize() + 2; y+=2)
+  {
+    for (int x = -2; x < qr.getSize() + 2; x++)
+    {
+      if (qr.getModule(x, y) && qr.getModule(x, y + 1))
+        std::cout << "\u2588";
+      else if (qr.getModule(x, y) && !qr.getModule(x, y + 1))
+        std::cout << "\u2580";
+      else if (!qr.getModule(x, y) && qr.getModule(x, y + 1))
```

Comment:
Calls to getModule could be cached if they're expensive. Don't bother it it's mostly just a lookup.

Reply:
Displaying a QR code is basically instant. I don’t think there would a noticeable difference. Would you still prefer to cache it?

Reply:
If it's ~instant for the user, no need.